Microsoft.TeamServices.Samples.Client/WorkItemTracking/QueriesSample.cs

```diff
@@ -12,24 +13,401 @@ public class QueriesSample : ClientSample
+            WorkItemTrackingHttpClient workItemTrackingClient = connection.GetClient<WorkItemTrackingHttpClient>();
+
+            QueryHierarchyItem query = workItemTrackingClient.GetQueryAsync(project, queryName, null, 2).Result;
+
+            if (query == null)
+            {
+                Console.WriteLine("No queries found for path '{0}'", queryName);
+            }
+            else
+            {
+                Console.WriteLine("Queries:");
+
+                foreach (var item in query.Children)
+                {
+                    Console.WriteLine("{0}", item.Name);
+                    Console.WriteLine("  {0}", item.Id);
+                    Console.WriteLine("  {0}", item.Path);
+                    Console.WriteLine();
+                }
+            }
+
+            return query;
+        }
+
+        [ClientSampleMethod]
+        public List<QueryHierarchyItem> GetListOfQueriesAndFoldersWithOptions()
+        {
+            System.Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+
+            VssConnection connection = Context.Connection;
+            WorkItemTrackingHttpClient workItemTrackingClient = connection.GetClient<WorkItemTrackingHttpClient>();
+
+            List<QueryHierarchyItem> queries = workItemTrackingClient.GetQueriesAsync(projectId, QueryExpand.All, 1).Result;
+
+            if (queries == null)
+            {
+                Console.WriteLine("No queries found");
+            }
+            else
+            {
+                Console.WriteLine("Queries:");
+
+                foreach (var query in queries)
+                {
+                    Console.WriteLine("  {0} - {1}", query.Name, query.Path);
+                }
+            }
+
+            return queries;
+        }
+
+        [ClientSampleMethod]
+        public List<QueryHierarchyItem> GetListOfQueriesAndFoldersIncludeDeleted()
+        {
+            System.Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+
+            VssConnection connection = Context.Connection;
+            WorkItemTrackingHttpClient workItemTrackingClient = connection.GetClient<WorkItemTrackingHttpClient>();
+
+            List<QueryHierarchyItem> queries = workItemTrackingClient.GetQueriesAsync(projectId, QueryExpand.None, 1, true).Result;
+
+            if (queries == null)
+            {
+                Console.WriteLine("No queries found");
+            }
+            else
+            {
+                Console.WriteLine("Queries:");
+
+                foreach (var query in queries)
+                {
+                    Console.WriteLine("  {0} - {1}", query.Name, query.Path);
+                }
+            }
+
+            return queries;
+        }
+
+        [ClientSampleMethod]
+        public QueryHierarchyItem GetQueryOrFolderById()
+        {
+            System.Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+            string queryId = "2614c4de-be48-4735-9fdc-9656f55c495f";
```

Comment:
Is this a well-known query ID? If so, say this in a /* xxxx */ comment. If not, can you have one of the earlier methods store a valid query ID into the Context so this API will have one to use?